src/components/cc-invoice/cc-invoice.js

```diff
@@ -32,49 +32,64 @@ export class CcInvoice extends LitElement {
 
   static get properties () {
     return {
-      error: { type: Boolean },
       invoice: { type: Object },
-      number: { type: String },
     };
   }
 
   constructor () {
     super();
 
-    /** @type {boolean} Sets a loading error state. */
-    this.error = false;
-
-    /** @type {Invoice|null} Sets the invoice. */
-    this.invoice = null;
+    /** @type {InvoiceState} Sets the invoice state. */
+    this.invoice = { state: 'loading' };
+  }
 
-    /** @type {string|null} Sets the invoice number. */
-    this.number = null;
+  _getInvoiceNumber () {
+    if (this.invoice.state === 'loading') {
+      return this.invoice.number;
+    }
+    if (this.invoice.state === 'error') {
+      return this.invoice.number;
+    }
+    if (this.invoice.state === 'loaded') {
+      return this.invoice.invoice.number;
+    }
+    return null;
   }
 
   render () {
 
-    const skeleton = (this.invoice == null);
-    const invoice = skeleton ? SKELETON_INVOICE : this.invoice;
-
-    const number = this.number ?? SKELETON_INVOICE.number;
-    const date = invoice.emissionDate;
-    const amount = invoice.total.amount;
+    const invoiceNumber = this._getInvoiceNumber();
+    const skeletonNumber = invoiceNumber == null;
+    const number = skeletonNumber ? SKELETON_INVOICE.number : invoiceNumber;
+    const skeleton = skeletonNumber || (this.invoice.state === 'loading');
+    const invoice = skeleton ? SKELETON_INVOICE : this.invoice.invoice;
+    const title = `${i18n('cc-invoice.title')} ${number}`;
```

Comment:
**suggestion:** I think we could simplify a little bit by doing this + some modifications within the HTML (see my next comment)
```suggestion
    const invoiceNumber = this._getInvoiceNumber();
    const skeleton = (this.invoice.state === 'loading');
    const skeletonNumber = invoiceNumber == null;
    const number = skeletonNumber ? SKELETON_INVOICE.number : invoiceNumber;
    const invoice = skeleton ? SKELETON_INVOICE : this.invoice.invoice;
```